### PR TITLE
[XPU] support glm-4.5-air (fix neox+partial_rotary_factor)

### PR DESCRIPTION
**Pull request by zhupengyang**
## Motivation

为 XPU 平台添加 GLM-4.5-air 模型支持，优化 MoE 算子和 RoPE 实现。
> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

## Modifications

1. **fused_noaux_tc 算子**：调整输出顺序为 `{gating_logits, topk_weights, topk_ids}`
2. **block_attn 算子**：添加 `rope_head_dim` 参数传递，优化 encoder 模式下的 qkv 指针偏移
3. **rotary_embedding**：XPU 平台使用 `current_platform.is_xpu()` 检测，优化 `GlmRotaryEmbedding` 实现
4. **glm4_moe.py**：修正 `MergedReplicatedLinear` 参数名 `output_size` → `output_sizes`
5. **linear.py**：添加 `output_dim` 的 `None` 检查
6. **xpu_model_runner.py**：`get_rope` 调用添加 `partial_rotary_factor` 参数

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

**Review by mayang002 — APPROVED** [review]

File: fastdeploy/model_executor/layers/quantization/__init__.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def parse_quant_config(args, model_config, is_ernie, is_v1_loader):
             logger.warning(f"Failed to parse quantization config normally ({e}), trying fallback")
             quant_config_name = args.quantization["quantization"]
             quantization_config["quantization"] = quant_config_name
+        model_config.quantization_config = quantization_config
```
Comment (zhupengyang) [inline fastdeploy/model_executor/layers/quantization/__init__.py:139]: 修复从--quantization参数传json形式的配置不生效的问题

File: fastdeploy/model_executor/layers/linear.py [hunk 2]
```diff
@@ -61,7 +61,8 @@ def create_weights(self, layer: nn.Layer, **extra_weight_attrs):
         )
 
         if self.model_format == "torch" and "output_dim" in extra_weight_attrs:
-            extra_weight_attrs["output_dim"] = not extra_weight_attrs["output_dim"]
+            if extra_weight_attrs["output_dim"] is not None:
+                extra_weight_attrs["output_dim"] = not extra_weight_attrs["output_dim"]
```
Comment (zhupengyang) [inline fastdeploy/model_executor/layers/linear.py:65]: 修复torch模型开ep+tp报错的问题
